### PR TITLE
Add skip method

### PR DESCRIPTION
**Pull request by bkw**
very handy for quickly ignoring a region of the buffer without copying any data.
To be applied after #4 (overlapping context)

### CODE REVIEW

File: test/unit/test-Reader.js [hunk 1]
```diff
@@ -242,6 +242,20 @@ test('Read Methods', {
     assert.deepEqual(reader.buffer(3), new Buffer([1, 2, 3]));
     assert.deepEqual(reader.buffer(2), new Buffer([4, 5]));
   },
+  'buffer: return a copy' : function () {
+    var reader = new Reader(new Buffer([1]));
+    var origResult = reader.buffer(1);
+    reader.write(new Buffer([2]));
+    assert.deepEqual(origResult, new Buffer([1]));
+  },
```
Comment (felixge) [inline test/unit/test-Reader.js:250]: Looks like this shouldn't be part of this PR

File: lib/Reader.js [hunk 2]
```diff
@@ -168,5 +168,14 @@ Reader.prototype._nullDistance = function() {
 
 Reader.prototype.buffer = function(bytes) {
   this._offset += bytes;
-  return this._buffer.slice(this._offset - bytes, this._offset);
+  var ret = new Buffer(bytes);
+  this._buffer.copy(ret, 0, this._offset -  bytes, this._offset);
+  return ret;
+};
```
Comment (felixge) [inline lib/Reader.js:174]: Looks like this shouldn't be part of this PR

File: README.md [hunk 3]
```diff
@@ -96,6 +96,11 @@ omitted, a null terminated string is assumed.
 
 Returns the next `bytes` as a buffer.
 
+### reader.skip([bytes])
```
Comment (felixge) [inline README.md:99]: Why the `[` and `]` ? I use this for indicating optional arguments, which doesn't seem to be the case here.

**Comment by felixge** [comment]
LGTM, except for the things I commented on. Can you update?

File: test/unit/test-Reader.js (hunk 1 above)
Comment (bkw) [inline test/unit/test-Reader.js:250]: yeah, that was beause the buffer-copy patch was not merged when I did this, and the patch context overlaps. I did'nt know how to supply a patch that would merge cleanly otherwise. I'll try to rebase, then this should go away.

File: README.md (hunk 3 above)
Comment (bkw) [inline README.md:99]: ahh, stupid me just copy&pasting from the line above. Will fix.

**Comment by bkw** [comment]
new pr: #6
